arbnode/seq_coordinator.go

```diff
@@ -492,6 +506,64 @@ func (c *SeqCoordinator) updateWithLockout(ctx context.Context, nextChosen strin
 	return c.noRedisError()
 }
 
+func (c *SeqCoordinator) deleteFinalizedMsgsFromRedis(ctx context.Context) error {
+	finalized, err := c.sync.GetFinalizedMsgCount(ctx)
+	if err != nil || finalized == 0 {
+		return fmt.Errorf("finalizedMessageCount is zero or error getting finalizedMessageCount from syncMonitor: %w", err)
+	}
+	updateFinalizedMsgCount := func() error {
+		finalizedBytes, err := c.msgCountToSignedBytes(finalized)
+		if err != nil {
+			return err
+		}
+		if err = c.Client.Set(ctx, redisutil.FINALIZED_MSG_COUNT_KEY, finalizedBytes, c.config.SeqNumDuration).Err(); err != nil {
+			return fmt.Errorf("couldn't set %s key to current finalizedMsgCount in redis: %w", redisutil.FINALIZED_MSG_COUNT_KEY, err)
+		}
+		return nil
+	}
+	prevFinalized, err := c.getRemoteFinalizedMsgCount(ctx)
+	if err != nil {
+		if errors.Is(err, redis.Nil) {
+			var keys []string
+			for msg := finalized; ; msg-- {
+				exists, err := c.Client.Exists(ctx, redisutil.MessageKeyFor(msg), redisutil.MessageSigKeyFor(msg)).Result()
+				if exists == 0 || err != nil {
+					break
+				}
+				keys = append(keys, redisutil.MessageKeyFor(msg), redisutil.MessageSigKeyFor(msg))
+			}
+			// If there is an error deleting finalized messages during init, we retry later either from this sequencer or from another
+			if len(keys) > 0 {
+				log.Info("Initializing finalizedMsgCount and deleting finalized messages from redis", "finalizedMsgCount", finalized)
+				if err := c.Client.Del(ctx, keys...).Err(); err != nil {
+					return fmt.Errorf("error deleting finalized message and their signatures from redis during init of finalizedMsgCount: %w", err)
+				}
+			}
+			return updateFinalizedMsgCount()
+		}
+		return fmt.Errorf("error getting finalizedMsgCount value from redis: %w", err)
+	}
+	remoteMsgCount, err := c.GetRemoteMsgCount()
+	if err != nil {
+		return fmt.Errorf("cannot get remote message count: %w", err)
+	}
+	msgToDelete := finalized
+	if msgToDelete > remoteMsgCount {
+		msgToDelete = remoteMsgCount
+	}
```

Comment:
nitpick: you can use msgToDelete := min(finalized, remoteMsgCount)

Reply:
ahh yes, go introduced this in recent version :) 
makes it so much easier!